### ISSUE TITLE
Is it normal to use 2 black and white images?

### ISSUE THREAD

**Issue by Keiser04** [junyanz/pytorch-CycleGAN-and-pix2pix#1610]
During the training with the colorize mode, something caught my attention and that is that during the training it uses two black and white images when one is in color.

This should be the norm, right?
![imagen](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/assets/146493916/808376ea-c55e-4e33-8b37-073180b94d9c)
But this....?
![imagen](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/assets/146493916/7244c46f-6399-40f1-aa0a-2192ae7fa1c6)
![imagen](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/assets/146493916/aab6d946-6409-4153-84c9-50070604c5a1)
![imagen](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/assets/146493916/6152a4d8-7052-4081-a098-ef0492b1f3f7)

**Comment by JustinasLekavicius** [comment]
Check what your input_nc and output_nc values are. 1 is for grayscale, 3 is for RGB.

**Comment by Keiser04** [comment]
is that I do not put those arguments

> Check what your input_nc and output_nc values are. 1 is for grayscale, 3 is for RGB.

**Comment by JustinasLekavicius** [comment]
> is that I do not put those arguments
> 
> > Check what your input_nc and output_nc values are. 1 is for grayscale, 3 is for RGB.
> 
> 

Check if you put those arguments in your training configuration. If your input images are grayscale, use input_nc=1. If the images are colored, use input_nc=3

**Comment by Keiser04** [comment]
> > is that I do not put those arguments
> > > Check what your input_nc and output_nc values are. 1 is for grayscale, 3 is for RGB.
> 
> Check if you put those arguments in your training configuration. If your input images are grayscale, use input_nc=1. If the images are colored, use input_nc=3

no, I didn't use them, I used the ones that came with the colab and with the one to resume training, but nothing else.

**Comment by Keiser04** [comment]
or maybe they have defaults?

**Comment by JustinasLekavicius** [comment]
> or maybe they have defaults?

It appears that for colorization dataset, the defaults are input_nc 1 and output_nc 2. So the issue may be somewhere else. T
Checking the code and data you use for training would be useful.

**Comment by Keiser04** [comment]
> > or maybe they have defaults?
> 
> It appears that for colorization dataset, the defaults are input_nc 1 and output_nc 2. So the issue may be somewhere else. T Checking the code and data you use for training would be useful.

If I change the output to 3 should it work?
